src/lib.rs

```diff
@@ -17,9 +17,8 @@ impl<T: Default> Trailer<T> {
     pub fn new(capacity: usize) -> Trailer<T> {
         unsafe {
             let trailer = Trailer::allocate(capacity);
-            let previous = mem::replace(&mut *(trailer.ptr as *mut T), T::default());
-            mem::forget(previous);
-
+            let ptr = trailer.ptr as *mut T;
+            ptr.write(T::default());
```

Comment:
ptr write is the write way to write into an invalid/uninitialized part of memory